### DISCUSSION TITLE
OFF TOPIC: Use of prototype.destroy = {}

### DISCUSSION THREAD

**Discussion by nielsnl68** [cronvel/terminal-kit#174] [General]
Hi sorry for asking and i never found a good source, i tried the W3school but the info i a bit to vague.

I know the existence of the <object>.prototype ....
But never found the right source to learn about the possibilities etc.
For instance you are using the prototype technic a lot. And i was wondering what your source to learn about possibilities of it ?

The reason is like your Element class i need to clean up some created events etc.
searching for it does explain the existence of it but nowhere about how it is used.

most apricated.
Niels

**Comment by cronvel** [comment]
Javascript used to be prototype-based object oriented language. Now it is possible to used the more common class-based in the latest EcmaScript standard. I'm still using prototype because of habit, and almost all my project predate the new class syntax.

To learn more about that, read [this article on MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object_prototypes) (MDN is better than w3school).

`Element` is a base class (a.k.a. a superclass), it is not meant to be instanciated, but everything inherit from it.

**Reply by nielsnl68** [reply]
thanks, i will take a read over there, it is maybe an old method but i see it heavily used sometimes, like in your classes.
Yes i understand that construction of the Element class. Very nice setup. i love this kind if setup.

I learned to work the OOP way back in the early 90ties. My backround is Pascal/delphi back in the days. Started with the Commodore 64 early 80ties. I did some visual Basic stuff but it was just not my thing. (I have seen beautiful thinks made with it). 
I really started to understand how OOP really worked was when my boss at that time bought at that time the latest version of Clipper (a programming language around a database named DBase also very old). That flipped my brain and since then i was hooked on OOP, i never want to go back to flat programming. Now a days i program in Java(Android), PHP (webservers), JavaScript (client side work), some C(++) (arduino/esp32) and i just started my first NodeJs application.

In that time a used and created many GUI frameworks for myself or for the bosses i worked for. Some where bad some where better. CSS was one of the bad once in my eyes, que positioning of objects/widged's etc. at other parts it better then others.
  
The last couple of days i have been looking on what you have made so far, and it does look amazingly good. There are some place where some tweaking could be added, but in the most part, hats off for you. 

And i love the new addition (document framework). For me it is the way to go.
But, just to say, when i had not looked into it. I never found out that you mend something totally differed then i trough it was, namely a way to show a document (textviewer). 
The idea to make it responsive is very nice and i love to use it.  For me is the positioning system a small bottle neck, because not all widget's are aware of the responsive design and setup the positioning at the begin and so far i could tell from the scripts i read. the positioning is calculated at creation, but i could be wrong.

Again, i would offer my time and knowledge to you to make this project the most outstanding terminal framework that there exists.

**Reply by cronvel** [reply]
Thanks for the praise ;)
